airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/LargeEnrollmentCallout.tsx

```diff
@@ -9,12 +9,17 @@ import { Text } from "components/ui/Text";
 
 import { ReactComponent as ConnectorsBadges } from "./connectors-badges.svg";
 import { useShowEnrollmentModal } from "./EnrollmentModal";
+import { useFreeConnectorProgram } from "./hooks/useFreeConnectorProgram";
 import styles from "./LargeEnrollmentCallout.module.scss";
 
 export const LargeEnrollmentCallout: React.FC = () => {
   const { showEnrollmentModal } = useShowEnrollmentModal();
+  const { userDidEnroll } = useFreeConnectorProgram();
 
-  return (
+  return userDidEnroll ? (
+    // eslint-disable-next-line react/jsx-no-useless-fragment
```

Comment:
If the whole UI subtree should be rendered or not based on some condition, I think keeping the conditional logic outside the UI it toggles, like this:
```tsx
return someCond ? (
  <></>
) : (
  <TheActualComponent>
    <And>
      <ItsChildren />
    </And>
  </TheActualComponent>
);
```
is _much_ nicer than sticking it inside the nominal UI it's trying to toggle:
```tsx
return (
  <>
    {someCond ? null : (
      <TheActualComponent>
        <And>
          <ItsChildren />
        </And>
      </TheActualComponent>
    )}
  </>
)
```